qt/scientific_interfaces/Indirect/IDAFunctionParameterEstimation.cpp

```diff
@@ -0,0 +1,34 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "IDAFunctionParameterEstimation.h"
+#include "MantidAPI/IFunction.h"
+
+#include <math.h>
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+// Add function name and estimation function to the stored function map.
+void IDAFunctionParameterEstimation::addParameterEstimationFunction(
+    const std::string &name, EstimationFunction function) {
```

Comment:
It seems Cpp check won't let me pass the string by value, so as far as I can tell I have to pass it be reference. But I've called `std::move` on the function to avoid the copying.

Reply:
Good point!